### DISCUSSION TITLE
PartialNonUniformLoad

### DISCUSSION THREAD

**Discussion by cmoha** [BriefFiniteElementNet/BriefFiniteElement.Net#109] [Q&A]
Dear Eng. Ehsan, I hope that you are having a good day.
Can you give some example on how we can use PartialNonUniformLoad as given in below image.
I have looked into one example but I could't figure what are the parameter of severity function .
![image](https://user-images.githubusercontent.com/10573467/105664812-40eda980-5ede-11eb-8409-e884aeecda17.png)

**Comment by epsi1on** [comment]
First of all, this type of load is still under development and can have bugs.

The image you did provide is a linear varying load, e.g. start to end can connected with a straight line:

![image](https://user-images.githubusercontent.com/7839141/105796674-98564d00-5fa4-11eb-9b4a-bbdcb86a1f22.png)

`PartialNonUniformLoad.SeverityFunction` is a property of type `IPolynomial`. A line is a polynomial itself with degree of `1` and can be expressed as `IPolynomial`. This is an example https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/blob/df828552370d0e53ef5098743ef8bb255fe70bfe/BriefFiniteElementNet.CodeProjectExamples/PartialTrapezoidalLoadExamples.cs#L15-L24

The severity function can be a linear varying also can be a Polynomial (for example quadratic or ...).

also the documentation in ReadTheDocs is updated

**Reply by cmoha** [reply]
to understand yo well, are these coordinates in
 load.SeverityFunction = Mathh.SingleVariablePolynomial.FromPoints(-1, 5, 1, 7); 
are passes as isoparametric coordinates or cartesian coordinates?

**Reply by epsi1on** [reply]
They are isoparametric, this code

`SingleVariablePolynomial.FromPoints(-1, 5, 1, 7);` 

Will create a non-uniform load with start value of 5 at x=0 (ξ=-1) and end value of 7 at x=L (ξ=+1)

**Reply by epsi1on** [reply]
Did it work?

**Reply by cmoha** [reply]
Tomorrow, I will test this load for bar element with different cases.

**Reply by cmoha** [reply]
Dear Ehsan,
I have tested many cases and in all cases, this type of load is not recognized and no reactions nor internal forces are calculated.
this is the simplest example that i have used.
`


 public void Run3()
        {
            var model = new Model();

            //adding nodes
            model.Nodes.Add(new Node(0, 0, 0) { Label = "n0" });
            model.Nodes.Add(new Node(5, 5, 5) { Label = "n1" });

            //adding elements
            model.Elements.Add(new BarElement(model.Nodes["n0"], model.Nodes["n1"]) { Label = "e0" });

            //assign constraint to nodes
            model.Nodes["n0"].Constraints = model.Nodes["n1"].Constraints = Constraints.MovementFixed;

            //define sections and material
            var secAA = new Sections.UniformGeometric1DSection(SectionGenerator.GetISetion(0.24, 0.67, 0.01, 0.006));
            var mat = Materials.UniformIsotropicMaterial.CreateFromYoungPoisson(210e9, 0.3);

            //assign materials
            (model.Elements["e0"] as BarElement).Material = mat;

            //assign sections
            (model.Elements["e0"] as BarElement).Section = secAA;

            //creating loads
            var u1 = new Loads.PartialNonUniformLoad() { Direction = Vector.K, CoordinationSystem = CoordinationSystem.Global };
            u1.SeverityFunction = Mathh.SingleVariablePolynomial.FromPoints(-1, -6, 1, -4);
            u1.StartLocation = new IsoPoint(-0.5);      //set locations of trapezoidal load
            u1.EndLocation = new IsoPoint(0.5);         //set locations of trapezoidal load

            //assign loads
            model.Elements["e0"].Loads.Add(u1);

            //solve model
            model.Solve_MPC();

            //retrieve solve result
            var n0reaction = model.Nodes["n0"].GetSupportReaction();
            var n1reaction = model.Nodes["n1"].GetSupportReaction();

            Console.WriteLine("Support reaction of n0: {0}", n0reaction);
            Console.WriteLine("Support reaction of n1: {0}", n1reaction);

            var x = 1.0; //need to find internal force at x = 1.0 m
            var iso = (model.Elements["e0"] as BarElement).LocalCoordsToIsoCoords(x);//find the location of 1m in iso coordination system
            var e4Force = (model.Elements["e0"] as BarElement).GetInternalForceAt(iso[0]);//find internal force
            Console.WriteLine("internal force at x={0} is {1}", x, e4Force);
        }`

**Comment by cmoha** [comment]
any updates?

**Reply by epsi1on** [reply]
added issue #111 
will check it
thanks
